optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/recaller/BestSolutionRecaller.java

```diff
@@ -108,19 +117,26 @@ public void processWorkingSolutionDuringMove(Score score, AbstractStepScope<Solu
         if (bestScoreImproved) {
             phaseScope.setBestSolutionStepIndex(stepScope.getStepIndex());
             Solution_ newBestSolution = solverScope.getScoreDirector().cloneWorkingSolution();
-            updateBestSolution(solverScope, score, newBestSolution);
+            updateBestSolutionAndFire(solverScope, score, newBestSolution);
         } else if (assertBestScoreIsUnmodified) {
             solverScope.assertScoreFromScratch(solverScope.getBestSolution());
         }
     }
 
-    public void updateBestSolution(SolverScope<Solution_> solverScope) {
+    public void updateBestSolutionAndFire(SolverScope<Solution_> solverScope) {
         Solution_ newBestSolution = solverScope.getScoreDirector().cloneWorkingSolution();
         Score newBestScore = solverScope.getSolutionDescriptor().getScore(newBestSolution);
-        updateBestSolution(solverScope, newBestScore, newBestSolution);
+        updateBestSolutionAndFire(solverScope, newBestScore, newBestSolution);
     }
 
-    protected void updateBestSolution(SolverScope<Solution_> solverScope, Score bestScore, Solution_ bestSolution) {
+    private void updateBestSolutionAndFire(SolverScope<Solution_> solverScope, Score bestScore, Solution_ bestSolution) {
+        // We do not want construction heuristics to fire intermediate best solution changed events.
+        // We still want best solution and score updated though, so that unimproved* terminations work correctly.
```

Comment:
I'd move these 2 comments in

Code style: comments shouldn't use "we" and should be written more directive (= shorter):
"We do not want construction heuristics to fire intermediate best solution changed events."
=>  "Construction heuristics don't fire intermediate best solution changed events."
"We still want best solution and score updated though, so that unimproved* terminations work correctly."
=> "But the best solution and score are updated, so that unimproved* terminations work correctly."

_Speak to peer coders as an Englishman, write code comments as a German._